redash/query_runner/mongodb.py

```diff
@@ -158,10 +178,12 @@ def __init__(self, configuration):
         )
 
     def _get_db(self):
+        kwargs = {}
         if self.is_replica_set:
             db_connection = pymongo.MongoClient(
                 self.configuration["connectionString"],
                 replicaSet=self.configuration["replicaSetName"],
+                readPreference=self.configuration.get("readPreference", "primaryPreferred")
```

Comment:
The UI will use the first value of the enum as the default, which in this case is `primary`. If you want to have `primaryPreferred` as the default, you need to have it as the first one in the list.

Reply:
Updated it to work like that.